apps/api/validators.py

```diff
@@ -1,12 +1,384 @@
-from django.core.exceptions import ValidationError
+'''Validates if the file given is a csv file'''
 
 import csv
+from re import match
+from django.core.exceptions import ValidationError
+
+
+class ColumnError(Exception):
+    '''Exception raised if the number of columns in the csv file header does
+    not match the rest'''
+    pass
+
+
+class MultipleUtoridOccurences(Exception):
+    '''Exception raised if the same utor id occurs multiple times in a file
+    '''
+    pass
+
+
+class InvalidUtorid(Exception):
+    '''Utor id not valid. It has to be 8 characters long and only contain
+    alphanumberic symbols
+    '''
+    pass
+
+
+class InvalidData(Exception):
+    '''If the data at a given column does not have the proper structure or
+    seem valid
+    '''
+    pass
+
+
+def _is_lowercase_alphabet(value):
+    '''Check if the given parameter is entirely lowercase
+
+    Attribute:
+        value: str
+
+    Returns:
+        A boolean that determines if the parameter 'value' is completely
+        lowercase or not
+    '''
+    return len(match('[a-z]+', value).group(0)) == len(value)
+
+
+def _is_alphabet(value):
+    '''Check if the given parameter is an alphabet either upper or lowercase
+
+    Attribute:
+        value: str
+
+    Returns:
+        A boolean that determines if the parameter 'value' is composed
+        entirely of alphabet characters
+    '''
+    return len(match('[A-Za-z]+', value).group(0)) == len(value)
+
+
+def _is_alphanumeric(value):
+    '''Check if the given parameter is composed of alphabet and numeric
+    characters
+
+    Attribute:
+        value: str
+
+    Returns:
+        A boolean that determines if the parameter 'value' is composed of
+        alphabet and numeric characters
+    '''
+    return len(match('[A-Za-z0-9]+', value).group(0)) == len(value)
+
+
+def _is_alphanumeric_and_symbols(value):
+    '''Check if the given parameter is composed of alphabet, numeric and
+    certain symbols
+
+    Attribute:
+        value: str
+
+    Returns:
+        A boolean that determines if the parameter 'value' is composed of
+        alphabet, numeric and certain symbol characters
+    '''
+    return len(match('[A-Za-z0-9_.!@#$%^&*()\' ]+', value).group(0)) == len(value)
+
+
+def _is_numeric(value):
+    '''Check if the given parameter is composed of numeric characters
+
+    Attribute:
+        value: str
+
+    Returns:
+        A boolean that determines if the parameter 'value' is composed of
+        numberic characters
+    '''
+    return len(match('[0-9]+', value).group(0)) == len(value)
+
+
+def _valid_utorid(value):
+    '''Check if the given parameter is a valid utorid
+
+    Attribute:
+        value: str
+
+    Returns:
+        A boolean that determines if the parameter 'value' is a valid utorid.
+        Utor id is 8 characters in length and composed of alphabet and number
+    '''
+    return len(value) == 8 and _is_alphanumeric(value)
+
+
+def _valid_student_number(value):
```

Comment:
Returns true if value is alphanumeric.

Possible solution:

	Use numeric(value) instead of alphanumeric(value)